blivet/devices/lvm.py

```diff
@@ -792,6 +792,10 @@ def _preDestroy(self):
         # set up the vg's pvs so lvm can remove the lv
         self.vg.setupParents(orig=True)
 
+        # setting up VG's PVs may have caused this LV was automatically
+        # activated, make sure it is deactivated before removal
+        self._teardown()
```

Comment:
I worry about 1) a race where the lv isn't activated until between the `lvdeactivate` and the `lvremove` and 2) an auto-activated format such as swap (which would be taken care of by calling the full `teardown`, assuming no races).